myapp/app/controllers/application_controller.rb

```diff
@@ -16,4 +16,10 @@ def redirect_to_root_path_if_logged_in
       redirect_to root_path
     end
   end
+
+  def redirect_to_root_path_if_normal_role
+    if is_normal?(current_user)
```

Comment:
Since you're using an Enum, I think you can write it like this. It doesn't seem necessary to create a helper method.
https://api.rubyonrails.org/v5.1/classes/ActiveRecord/Enum.html
```suggestion
    if current_user.role_normal?
```

Reply:
Thank you. it is good to know, I fixed the code.